src/build/__init__.py

```diff
@@ -133,6 +145,23 @@ def _invoke_wrapped_runner(cmd: Sequence[str], cwd: str | None, extra_environ: M
     return _invoke_wrapped_runner
 
 
+def _parse_project_name(pyproject_toml: Mapping[str, Any]) -> str | None:
+    if 'project' not in pyproject_toml:
+        return None
+
+    project_table = dict(pyproject_toml['project'])
+
+    # If [project] is present, it must have a ``name`` field (per PEP 621)
+    if 'name' not in project_table:
+        return None
```

Comment:
This is an error condition.

Reply:
Do we want this to trigger a fatal error? Returning `None` here means we fallback to pulling the project name from the filename instead as if there was no `project` entry at all.

Reply:
Yeah, I think we should error if the project table is invalid - if it isn't rejected by us it'll be (or should be) rejected by the backend.

Reply:
Changed this to an exception, looks like the `test_build_with_dep_on_console_script` test `project` had no `name` field so I added one there as well.